### ISSUE TITLE
HttpClient per request

### ISSUE THREAD

**Issue by gerbendv** [Viincenttt/MollieApi#65]
I was about to start using the Mollie API with this project but then I noticed the usage of a seperate HttpClient instance for every request. Well, not exactly for every request but at least for every API-client by calling `this.CreateHttpClient(clientId, clientSecret);`.

To my knowledge this is not in accordance with best practices for using the HttpClient. This way of using HttpClient can lead to a lot of persistent open connections on the server running the application as well as a lot of overhead and CPU usage (for disposing the client). More about this can be read on https://blogs.msdn.microsoft.com/shacorn/2016/10/21/best-practices-for-using-httpclient-on-services/ and https://msdn.microsoft.com/en-us/library/system.net.http.httpclient(v=vs.110).aspx#Anchor_5.

Would it be an option to create one static instance that would stay alive during application lifetime? I myself usually use a singleton which is injected with ninject in my asp.net applications (`kernel.Bind<HttpClient>().ToSelf().InSingletonScope();`). This could be used as well but might be unhandy for folks not using dependency injection.

Any suggestions?

**Comment by gerbendv** [comment]
Hi Vincenttt,

I see the problem with the headers that would or would not be present on the HttpClient. A constructor override might be a good solution. I noticed that HttpClient has a overload of SendAsync in which an instance of HttpRequestMessage can be passed. That would enable you to add the right headers to the HttpRequestMessage.Headers collection.

**Comment by Viincenttt** [comment]
Hi gerbendv,

I just released a new version of the Mollie API (2.0.1.0) that allows you to pass your own HttpClient instance in the constructor of all the Mollie Client classes. 

For example:
```c#
HttpClient myHttpClientInstance = new HttpClient();
PaymentClient paymentClient = new PaymentClient(this.ApiTestKey, myHttpClientInstance);
PaymentRequest paymentRequest = new PaymentRequest() {
    Amount = new Amount(Currency.EUR, "100.00"),
    Description = "Description",
    RedirectUrl = this.DefaultRedirectUrl
};
PaymentResponse result = await paymentClient.CreatePaymentAsync(paymentRequest);
```

Please let me know if you have any other suggestions/ideas. 

Kind regards,
Vincent

**Comment by gerbendv** [comment]
Great, thanks!
